File: RefugeUA.WebApp/refugeua.webapp.client/src/app/core/models/accomodation-announcement.ts
import { Announcement } from "./announcement"
import { Image } from "./image"

export interface AccomodationAnnouncement extends Announcement{
    images?: Image[],
    buildingType: string,
    petsAllowed: boolean,
    floors: number,
    numberOfRooms: number,
    capacity: number,
    areaSqMeters?: number,
    price?: number,
    isFree: boolean
}

File: RefugeUA.Entities/AccomodationAnnouncement.cs
using RefugeUA.Entities.Abstracts;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RefugeUA.Entities
{
    public class AccomodationAnnouncement : Announcement
    {
        public ICollection<Image> Images { get; set; } = default!;

        public string BuildingType { get; set; } = default!;

        public bool PetsAllowed { get; set; }

        public ushort Floors { get; set; }

        public ushort NumberOfRooms { get; set; }

        public ushort Capacity { get; set; }

        public float? AreaSqMeters { get; set; }

        public float? Price { get; set; }

        public bool IsFree { get; set; }
    }
}
